.github/actions/bpftrace/scripts/check-encryption-leaks.bt

```diff
@@ -234,7 +243,8 @@ kprobe:br_forward
             $skb->dev->nd_net.net->ns.inum,
             $src_is_pod, $src_is_internal,
             $dst_is_pod, $dst_is_internal,
-            $pod_to_pod_via_proxy);
+            $pod_to_pod_via_proxy >= SK_TRACK_FLAG_BASE,
```

Comment:
so, if we have a `$pod_to_pod_via_proxy` whose value is SK_TRACK_FLAG_MASQUERADE(2), according to the hooks and logic, this packet didn't come from tcp_connect/udp_sendmsg but passed through __dev_queue_xmit? So it was not from proxy? but the output here is `proxy: 1(1)`?

Reply:
The packet still came from `tcp_connect/udp_sendmsg` as the map lookup succeeded!
1. In case the lookup fails, packet doesn't come from proxy
2. In case the lookup succeed:
    1. if value is `SK_TRACK_FLAG_BASE | SK_TRACK_FLAG_MASQUERADE`, then the packet has hit `tcp_connect` and `dev_xmit` without s-masquerading
    2. if value is only `SK_TRACK_FLAG_MASQUERADE`, the packet has hit `tcp_connect`, but then it was s-masqueraded, therefore a new entry in the map with only the value `SK_TRACK_FLAG_MASQUERADE` has been created (still it is coming from the proxy).

Reply:
> what i don't understand is, by the time br_forward is triggered, the packets must have passed __dev_queue_xmit@eth0, no? This flag will always be set?

Yes, always set, the difference is on which entry it is being set :smiley:
If no s-masq, then the lookup reuse the same entry, value will have both flags set.
If s-masq, then the lookup will reference a new entry with only this flag set.